### ISSUE TITLE
native-image-agent not respecting filter config for serialization-config.json

### ISSUE THREAD

**Issue by mitchelladam** [oracle/graal#3312]
**Describe the issue**
When using the native-image-agent to generate the json configuration files, access and caller filters are not being respected with regards to serialization-config.json. neither blanket nor specific excludeClasses work

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. git clone https://github.com/mitchelladam/graalvm-example.git
2. docker build . -t graal-example
3. docker run -it -v <path-to-graalvm-example>:/temp graal-example /bin/bash
4. cd /temp/
5. mvn clean package -P surefire-agent
6. see output in ./target/classes/META-INF/native-image/serialization-config.json contains a reference to org.junit even though org.junit.** is excluded in both caller-filter-file and access-filter-file

**Describe GraalVM and your environment:**
- Graalvm-ce latest docker image

**More details**
serializarion-config.json
```
[
  {
  "name":"org.junit.runner.Result$SerializedForm"
  }
]
```

filter
```
{ "rules": [
  {"excludeClasses": "junit.**"},
  {"excludeClasses": "org.junit.**"},
  {"excludeClasses": "org.apache.maven.surefire.**"},
  {"excludeClasses": "oracle.test.**"}
]
}
```

**Comment by kristofdho** [comment]
Can you verify on a dev build? I think this should be fixed by https://github.com/oracle/graal/pull/3268

**Comment by mitchelladam** [comment]
confirmed fixed in 21.1.0-dev-20210325_0249
see: https://github.com/mitchelladam/graalvm-example/tree/dev-build
